modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -173,63 +273,83 @@ public TableManager(
+            )
+        );
 
-        try {
-            Files.createDirectories(storageDir);
-        } catch (IOException e) {
-            throw new IgniteInternalException(
-                "Failed to create partitions store directory for " + name + ": " + e.getMessage(),
-                e
-            );
-        }
+        CompletableFuture.allOf(partitionsGroupsFutures.toArray(CompletableFuture[]::new)).thenRun(() -> {
+            try {
+                HashMap<Integer, RaftGroupService> partitionMap = new HashMap<>(partitions);
 
-        for (int p = 0; p < partitions; p++) {
-            RocksDbStorage storage = new RocksDbStorage(
-                storageDir.resolve(String.valueOf(p)),
-                ByteBuffer::compareTo
-            );
-
-            partitionsGroupsFutures.add(raftMgr.prepareRaftGroup(
-                raftGroupName(tblId, p),
-                assignment.get(p),
-                new PartitionListener(storage)
-            ));
-        }
+                for (int p = 0; p < partitions; p++) {
+                    CompletableFuture<RaftGroupService> future = partitionsGroupsFutures.get(p);
 
-        CompletableFuture.allOf(partitionsGroupsFutures.toArray(CompletableFuture[]::new)).thenRun(() -> {
-            HashMap<Integer, RaftGroupService> partitionMap = new HashMap<>(partitions);
+                    assert future.isDone();
 
-            for (int p = 0; p < partitions; p++) {
-                CompletableFuture<RaftGroupService> future = partitionsGroupsFutures.get(p);
+                    RaftGroupService service = future.join();
 
-                assert future.isDone();
+                    partitionMap.put(p, service);
+                }
 
-                RaftGroupService service = future.join();
+                InternalTableImpl internalTable = new InternalTableImpl(name, tblId, partitionMap, partitions);
 
-                partitionMap.put(p, service);
-            }
+                var schemaRegistry = new SchemaRegistryImpl(v -> schemaDesc);
 
-            InternalTableImpl internalTable = new InternalTableImpl(name, tblId, partitionMap, partitions);
+                schemaRegistry.onSchemaRegistered(schemaDesc);
+
+                var table = new TableImpl(
+                    internalTable,
+                    schemaRegistry,
+                    TableManager.this,
+                    null
+                );
 
-            var table = new TableImpl(internalTable, schemaReg, this, null);
+                tables.put(name, table);
+                tablesById.put(tblId, table);
 
-            tables.put(name, table);
-            tablesById.put(table.tableId(), table);
+                fireEvent(TableEvent.CREATE, new TableEventParameters(table), null);
 
-            onEvent(TableEvent.CREATE, new TableEventParameters(table), null);
+                Optional.ofNullable(createTblIntention.get(tblId)).ifPresent(f -> f.complete(table));
+            }
+            catch (Exception e) {
```

Comment:
Need to fire create table event with the exception here.

Reply:
Fixed.